src/Pos/Context/Functions.hs

```diff
@@ -43,8 +48,10 @@ import           Pos.Util.Context       (HasContext, askContext)
 genesisUtxoM :: (Functor m, WithNodeContext ssc m) => m Utxo
 genesisUtxoM = ncGenesisUtxo <$> getNodeContext
 
-genesisLeadersM :: (Functor m, WithNodeContext ssc m) => m SlotLeaders
-genesisLeadersM = ncGenesisLeaders <$> getNodeContext
+genesisLeadersM :: (MonadDB m, WithNodeContext ssc m) => m SlotLeaders
+genesisLeadersM = ifM isBootstrapEra
+    (pure $ NE.fromList $ HM.keys genesisBalances)
```

Comment:
I suppose `genesisBalances` is non-empty `Map`. But I would like to have non-empty list of genesis balances, from which `HashMap` is created. It will avoid using partial `NE.fromList` function.